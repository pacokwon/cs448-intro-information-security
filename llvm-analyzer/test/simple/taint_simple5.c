#include "homework.h"

int main() {
	int x = source(); 
	int y = 10;
	int result;

	if(x + y < 0){
		result = y * 10;
		sink(result);
	}
	else if(x + y < 10){
		result = x * y;
		sink(result);
	}
	else if(x + y < 100){
		result = sanitizer(x);
		sink(result);
	}
	else{
		result = x * y;
		sink(result);
	}
	return 0;
}
