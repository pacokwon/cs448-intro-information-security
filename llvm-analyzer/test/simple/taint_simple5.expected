Potential Tainted-flow @ simple/taint_simple5.c:main:14:3 (call void @sink(i32 %mul4), !dbg !30)
Potential Tainted-flow @ simple/taint_simple5.c:main:22:3 (call void @sink(i32 %mul11), !dbg !42)
